### DISCUSSION TITLE
uploadHandler on React router v7

### DISCUSSION THREAD

**Discussion by kmvan** [remix-run/remix#10268] [Q&A] [answered]
```
// missing
import {
  unstable_composeUploadHandlers,
  unstable_createFileUploadHandler,
  unstable_createMemoryUploadHandler,
  unstable_parseMultipartFormData,
} from 'react-router';
```

It looks like react router does not include these functions.

**Comment by gustavopch** [comment, marked as answer]
Those were removed. I believe using `@mjackson/form-data-parser` is the recommended approach. It's mentioned in the docs: https://reactrouter.com/how-to/file-uploads

**Reply by kmvan** [reply]
Thanks a lot.

**Reply by Longvu1982** [reply]
@gustavopch Hi how do we prevent a page reload when using encType="multipart/form-data"
